shardingsphere-db-protocol/shardingsphere-db-protocol-mysql/src/main/java/org/apache/shardingsphere/db/protocol/mysql/constant/MySQLServerInfo.java

```diff
@@ -34,27 +38,38 @@ public final class MySQLServerInfo {
     
     public static final MySQLCharacterSet DEFAULT_CHARSET = MySQLCharacterSet.UTF8MB4_GENERAL_CI;
     
-    private static final String DEFAULT_MYSQL_VERSION = "5.7.22";
+    private static final String DEFAULT_MYSQL_VERSION = "5.6.23";
     
     private static final String SERVER_VERSION_PATTERN = "%s-ShardingSphere-Proxy %s";
     
-    private static volatile String serverVersion;
+    private static Map<String, String> serverVersions = new ConcurrentHashMap<>();
     
     /**
      * Set server version.
-     *
+     * 
+     * @param schemaName schema name
      * @param serverVersion server version
      */
-    public static synchronized void setServerVersion(final String serverVersion) {
-        MySQLServerInfo.serverVersion = null == serverVersion ? null : String.format(SERVER_VERSION_PATTERN, serverVersion, CommonConstants.PROXY_VERSION.get());
+    public static void setServerVersion(final String schemaName, final String serverVersion) {
+        serverVersions.put(schemaName, String.format(SERVER_VERSION_PATTERN, serverVersion, CommonConstants.PROXY_VERSION.get()));
```

Comment:
A connection may have no schema name before client executing `use database`, this may cause NPE.